.github/workflows/release.yml

```
@@ -0,0 +1,222 @@
+---
+# Releases the agent
+name: release
+
+on:
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: 'The branch to release'
+        required: true
+        default: 'main'
+      version:
+        description: 'The version to release (e.g. 1.2.3). This workflow will automatically perform the required version bumps'
+        required: true
+      update_changelog:
+        description: |
+          If enabled, everything in the changelog from the "Unreleased" section will be automatically moved to a new section for the new release.
+          If disabled, the changelog needs to be prepared for the release manually before triggering this workflow.
+        type: boolean
+        required: true
+        default: true
+      dry_run:
+        description: If set, run a dry-run release
+        default: false
+        type: boolean
+
+env:
+  JAVA_VERSION: 17
+  JAVA_DIST: adopt
+
+permissions:
+  contents: write
+
+jobs:
+  prepare_release:
+    permissions:
+      contents: write
+    name: "Changelog and Version Bump"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.branch }}
+      - uses: elastic/apm-pipeline-library/.github/actions/setup-git@current
+      - name: Set up JDK ${{ env.JAVA_VERSION }}
+        uses: actions/setup-java@v3
+        with:
+          java-version: ${{ env.JAVA_VERSION }}
+          distribution: ${{ env.JAVA_DIST }}
+          cache: 'maven'
+      - name: Prepare changelog for release
+        if: ${{ inputs.update_changelog }}
+        run: |
+          java .ci/ReleaseChangelog.java CHANGELOG.asciidoc ${{ inputs.version }}
+          git commit -m "Prepare changelog for release ${{ inputs.version }}" CHANGELOG.asciidoc
+      - name: Bump version and add git tag
+        run: ./mvnw release:prepare -B -DpushChanges=false "-Darguments=-DskipTests -Dmaven.javadoc.skip=true" -DreleaseVersion=${{ inputs.version }}
+      - run: git push --atomic origin ${{ inputs.branch }} v${{ inputs.version }}
+
+
+  maven_central_deploy:
+    name: "Deploy to Maven Central (Buildkite)"
+    runs-on: ubuntu-latest
+    needs:
+      - prepare_release
+    steps:
+      - id: buildkite
+        name: Run Deploy
+        uses: reakaleek/apm-pipeline-library/.github/actions/buildkite@feature/buildkite-download-artifacts
```

```suggestion
        uses: elastic/apm-pipeline-library/.github/actions/buildkite@current
```

as soon as https://github.com/elastic/apm-pipeline-library/pull/2233 is merged and released

---

Released in https://github.com/elastic/apm-pipeline-library/releases/tag/v1.1.398